src/segmentedfilesystem/PacketManager.java

```diff
@@ -0,0 +1,68 @@
+package segmentedfilesystem;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class PacketManager {
+
+    File file1 = new File();
+    File file2 = new File();
+    File file3 = new File();
+
+    public boolean allPackagesReceived(){
+        return file1.fileComplete() && file2.fileComplete() && file3.fileComplete();
+    }
+
+    public void store(PacketStructure packet){
+        if(file1.datapackets.isEmpty()){
+            file1.add(packet);
+        }
+        else if(file2.datapackets.isEmpty()){

```

Comment:
I don't know for 100% sure, but I _think_ that you wanted to check whether this packet matched the ID of the previous file before you added it to `file2`. It's super possible (highly likely, actually) that the first three packets all come from the same file, and here you'll spread them out across all three `File` objects.